File: professional/Vaadin14-UsingTemplates/exercises/src/main/java/com/vaadin/training/template/exercises/ex3/ToDoView.java
package com.vaadin.training.template.exercises.ex3;

import com.vaadin.flow.component.Tag;
import com.vaadin.flow.component.dependency.JsModule;
import com.vaadin.flow.component.dependency.NpmPackage;
import com.vaadin.flow.component.polymertemplate.EventHandler;
import com.vaadin.flow.component.polymertemplate.PolymerTemplate;
import com.vaadin.flow.router.Route;
import com.vaadin.training.template.exercises.MainLayout;

import java.util.List;

@Tag("todo-view")
@NpmPackage(value="@polymer/paper-input", version = "3.0.2")
@JsModule("./src/todo-view.js")
@Route(value = ToDoView.ROUTE, layout = MainLayout.class)
public class ToDoView extends PolymerTemplate<ToDoModel> {
	public static final String ROUTE = "ex3";
	public static final String TITLE = "ToDo List";

	//TODO: add a server side method for the on-change event of paper input element.
	// the method should create a new ToDo, add it to the list and then clear the text of paper-input.
	@EventHandler
	void handleAdd() {
		ToDo toDo = new ToDo();
		toDo.setText(getModel().getText());
		getModel().getToDoList().add(toDo);
		getModel().setText("");
	}
}

File: professional/Vaadin14-UsingTemplates/exercises/frontend/src/todo-view.js
import { PolymerElement } from '@polymer/polymer/polymer-element.js';
import '@vaadin/vaadin-checkbox/vaadin-checkbox.js';
import '@polymer/paper-input/paper-input.js';
import { html } from '@polymer/polymer/lib/utils/html-tag.js';
class ToDoView extends PolymerElement {
    static get template() {
        return html `
   <div>  
    <!-- bind the 'value' property of paper-input to a String, like 'newText' in the Model --> 
    <!-- add a server side listener method for the 'on-change' event -->
    <paper-input placeholder="Enter your task here" value="{{text}}" on-change="handleAdd"></paper-input> 
   </div> 
   <!-- bind the 'items' property of the element to a list defined in the Model--> 
   <template is="dom-repeat" items="[[toDoList]]"> 
    <div> 
     <!-- create a vaadin-checkbox element and bind its 'value' to 'done' property of the item -->
        <vaadin-checkbox value="{{item.done}}"></vaadin-checkbox>
     <!-- show the 'text' of the item next to the checkbox --> 
        <span>[[item.text]]</span>
    </div> 
   </template> 
`;
    }

    static get is() {
        return 'todo-view'
    }
}
customElements.define(ToDoView.is, ToDoView);